python/plugins/processing/algs/gdal/fillnodata.py

```diff
@@ -136,7 +142,11 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         arguments.append(str(self.parameterAsInt(parameters, self.BAND, context)))
 
         if self.parameterAsBoolean(parameters, self.NO_MASK, context):
-            arguments.append('-nomask')
+            if GdalUtils.version() >= 3040000:
+                # The -nomask option is no longer supported since GDAL 3.4.0 https://github.com/OSGeo/gdal/pull/4201
+                feedback.pushInfo(self.tr('The -nomask option is no longer supported since GDAL 3.4.0'))
```

Comment:
This should raise a QgsProcessingException, so that eg someone running an old process or model which relied on this is explicitly advised that something has changed.

Reply:
Right! Done.